packages/web/src/components/spa/HackAUBG/TopButton/TopButton.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+import './top_button.css';
+
+export const TopButton = () => {
+    const [visible, setVisibility] = useState(false);
+
+    const toggleVisibility = () => {
+        const scrolled = document.documentElement.scrollTop;
+
+        if (scrolled > 300) {
+            setVisibility(true);
+        } else if (scrolled <= 300) {
+            setVisibility(false);
+        }
+    };
+
+    const scrollToTop = () => {
+        window.scrollTo({
+            top: 0,
+            behavior: 'smooth'
+        });
+    };
+
+    window.addEventListener('scroll', toggleVisibility);
```

Comment:
Can you add a comment here, something like `// this is executed every time there is a scrolling event/action` or something similar so that other devs know what it does. Most of the dev dep is not familiar with eventListeners yet.